File: datamodel/src/main/java/datamodel/clinical/VitalFunctionCategory.java
package datamodel.clinical;

import org.jetbrains.annotations.NotNull;

public enum VitalFunctionCategory {
    NON_INVASIVE_BLOOD_PRESSURE("Non-invasive blood pressure"),
    ARTERIAL_BLOOD_PRESSURE("Arterial blood pressure"),
    HEART_RATE("Heart rate"),
    SPO2("SpO2");

    @NotNull
    private final String display;

    VitalFunctionCategory(@NotNull final String display) {
        this.display = display;
    }

    @NotNull
    public String display() {
        return display;
    }
}

File: algo/src/main/kotlin/com/hartwig/actin/clinical/datamodel/VitalFunctionCategory.kt
package com.hartwig.actin.clinical.datamodel

import com.hartwig.actin.Displayable

enum class VitalFunctionCategory(private val display: String) : Displayable {
    NON_INVASIVE_BLOOD_PRESSURE("Non-invasive blood pressure"),
    ARTERIAL_BLOOD_PRESSURE("Arterial blood pressure"),
    HEART_RATE("Heart rate"),
    SPO2("SpO2"),
    OTHER("Other");

    override fun display(): String {
        return display
    }
}
